android/src/main/java/io/ably/lib/push/ActivationStateMachine.java

```diff
@@ -415,12 +412,53 @@ public void execute(HttpScheduler http, Callback<Void> callback) throws AblyExce
 				@Override
 				public void onSuccess(Void response) {
 					Log.i(TAG, "updated registration " + device.id);
-					machine.handleEvent(new ActivationStateMachine.RegistrationUpdated());
+					machine.handleEvent(new RegistrationSynced());
 				}
 				@Override
 				public void onError(ErrorInfo reason) {
 					Log.e(TAG, "error updating registration " + device.id + ": " + reason.toString());
-					machine.handleEvent(new ActivationStateMachine.UpdatingRegistrationFailed(reason));
+					machine.handleEvent(new SyncRegistrationFailed(reason));
+				}
+			});
+		}
+	}
+
+	private static void validateRegistration(final ActivationStateMachine machine) {
+		final LocalDevice device = machine.activationContext.getLocalDevice();
+		boolean useCustomRegistrar = machine.activationContext.getPreferences().getBoolean(ActivationStateMachine.PersistKeys.PUSH_CUSTOM_REGISTRAR, false);
+		if (useCustomRegistrar) {
```

Comment:
This isn't in the spec changes. Maybe the new spec should look like RSH3b3a/RSH3b3b?

> (RSH3b3a) If a custom registerCallback was provided to Push#activate, pass it the local DeviceDetails updated with the push details.
> (RSH3b3b) Otherwise, make an asynchronous HTTP POST request to /push/deviceRegistrations using the local DeviceDetails updated with the push details as body.

Reply:
+1, see https://github.com/ably/docs/pull/786/commits/cc8e7def1b356368b5d8601a1f8c9fad6bdb5145